src/util.rs

```diff
@@ -119,9 +102,19 @@ pub(crate) fn check_policy_stage<T: Serialize>(
     policy_stage: PolicyStage,
     policy_file: &str,
     input: &T,
+    data: &[T],
```

Comment:
Nit: If you want data to actually be generic, you need to use a different type.  Right now input *AND* the array of data must be the same type.

At a minimum, you should have:

```
fn check_policy_stage<T: Serialize, U:Serialize>(
    policy_engine: &mut Engine,
    policy_stage: PolicyStage,
    policy_file: &str,
    input: &T,
    data: &[U],
```

However, if you want data to *not* be monomorphic, you'd need to use an HList (likely k-list) or tuple-type structure to preserve the original types.

I don't think this is what you want at all.  For now, at least, I'd use a different generic for data. In the future, you may want to use a different type to encapsulate serialization more generically.

Reply:
Nice catch. I made the changes.